### PR TITLE
Creating deterministic augmentation IDs

### LINKED ISSUE #535: Update Augmentation - Deterministic Ids

### PR DESCRIPTION
**Pull request by robertandremitchell**
## Description

Changes eicr_augmentation.py to have deterministic IDs, updates subsequent code and tests.

## Related Issues

Closes #535 

## Additional Notes

[Add any additional context or notes that reviewers should know about.]

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist

Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers

Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

**Comment by BradySkylight** [comment]
As part of this PR we should also just grab the versionNumber of the incoming eicr and use that as our 'augmented' eicr version as well.

**Comment by robertandremitchell** [comment]
> As part of this PR we should also just grab the versionNumber of the incoming eicr and use that as our 'augmented' eicr version as well.

thanks, @BradySkylight ! Should be fixed now, but curious if you think our fallback should be hardcoding to versionNumber=1 versus raising an error? Right now I added a ValueError, but idk if we expect that this would actually trip up eICRs

**Comment by BradySkylight** [comment]
> > As part of this PR we should also just grab the versionNumber of the incoming eicr and use that as our 'augmented' eicr version as well.
> 
> thanks, @BradySkylight ! Should be fixed now, but curious if you think our fallback should be hardcoding to versionNumber=1 versus raising an error? Right now I added a ValueError, but idk if we expect that this would actually trip up eICRs

I think defaulting to 1 if there isn't a version number present makes sense.

**Comment by robertandremitchell** [comment]
> > > As part of this PR we should also just grab the versionNumber of the incoming eicr and use that as our 'augmented' eicr version as well.
> > 
> > 
> > thanks, @BradySkylight ! Should be fixed now, but curious if you think our fallback should be hardcoding to versionNumber=1 versus raising an error? Right now I added a ValueError, but idk if we expect that this would actually trip up eICRs
> 
> I think defaulting to 1 if there isn't a version number present makes sense.

makes sense, updated!

File: packages/augmentation/src/augmentation/services/eicr_augmenter.py [hunk 1]
```diff
@@ -197,10 +209,12 @@ def _get_new_effective_time(self) -> Element:
...
         """Generate a versionNumber element for the augmented eICR document."""
+        original_version_number = self._get_original_by_xpath("/ClinicalDocument/versionNumber")
+        original_version_number_value = (
+            original_version_number.get("value") or "1"
+        )  # default to "1" if value attribute is missing to avoid errors
         version_number_tag = etree.Element("versionNumber")
-        # hard code to 1 for now
-        # TODO: we may need to have some way to increment this later
-        version_number_tag.set("value", "1")
+        version_number_tag.set("value", original_version_number_value)
```
Comment (nickclyde) [inline packages/augmentation/src/augmentation/services/eicr_augmenter.py]: `_get_old_version_number()` already does `self._get_original_by_xpath("/ClinicalDocument/versionNumber")` (line 179), so we could call that here to avoid the duplication.

File: packages/augmentation/tests/unit/test_eicr_augmenter.py [hunk 2]
```diff
@@ -157,19 +148,48 @@ def test_get_old_document_id_preserves_assigning_authority_name_when_present(sel
...
+
+        assert version_number.get("value") == "1"
+
+    def test_generates_same_augmented_ids_for_same_seed(self):
+        """Tests deterministic augmented identifiers use stable values for the same seed."""
+        first_augmenter = EICRAugmenter(BASIC_ECR, [])
+        second_augmenter = EICRAugmenter(BASIC_ECR, [])
+
+        assert first_augmenter.new_doc_id == second_augmenter.new_doc_id
+        assert first_augmenter.new_set_id == second_augmenter.new_set_id
```
Comment (nickclyde) [inline packages/augmentation/tests/unit/test_eicr_augmenter.py]: I think the test name might be a bit misleading, since neither augmenter is given a seed, this is really testing "no-seed callers default to the same value when the XML is the same." To actually test "same seed produces same IDs", could we pass an explicit `deterministic_id_seed` (and maybe even use two different XML payloads with the same seed, to prove the seed, not the document, is what determines the ID)?

File: packages/augmentation-lambda/src/augmentation_lambda/lambda_function.py [hunk 3]
```diff
@@ -124,6 +124,7 @@ def _process_record(record: SQSRecord, s3_client: BaseClient) -> None:
             document=original_eicr,
             nonstandard_codes=augmenter_input.nonstandard_codes,
             config=config,
+            deterministic_id_seed=augmenter_input.eicr_id,
```
Comment (nickclyde) [inline packages/augmentation-lambda/src/augmentation_lambda/lambda_function.py]: Just to confirm my understanding: `augmenter_input.eicr_id` here is the S3 persistence_id (e.g. `2025/09/03/1-5f84c7a5-...`), not the eICR's `<id root>` UUID, right?

**Review by nickclyde — COMMENTED** [review]
Nice work Rob!!

File: packages/augmentation-lambda/src/augmentation_lambda/lambda_function.py (hunk 3 above)
Comment (robertandremitchell) [reply packages/augmentation-lambda/src/augmentation_lambda/lambda_function.py]: yeah, the augmenter_input.eicr_id is the S3 persistence_id parsed from the TTC output object key, not the eICR <id root>. My thought was the persistence_id as the deterministic seed would ensure IDs are stable.

File: packages/augmentation/tests/unit/test_eicr_augmenter.py (hunk 2 above)
Comment (robertandremitchell) [reply packages/augmentation/tests/unit/test_eicr_augmenter.py]: yeah, that makes sense. I've renamed it for clarity and added the additional test tha ttests the basic vs. covid eicr with the deterministic ID.

File: packages/augmentation/src/augmentation/services/eicr_augmenter.py (hunk 1 above)
Comment (robertandremitchell) [reply packages/augmentation/src/augmentation/services/eicr_augmenter.py]: ack, good call. updated!

File: packages/augmentation-lambda/src/augmentation_lambda/lambda_function.py (hunk 3 above)
Comment (JNygaard-Skylight) [reply packages/augmentation-lambda/src/augmentation_lambda/lambda_function.py]: Doesn't have to be in this PR, but that attribute should be renamed as in every other context eICR ID would mean the value of the ID element.

Comment (robertandremitchell) [reply packages/augmentation-lambda/src/augmentation_lambda/lambda_function.py]: totally makes sense. updated!

**Review by nickclyde — APPROVED** [review]
Great work Rob!
